core/blockchain.go

```diff
@@ -1713,20 +1715,20 @@ func (bc *BlockChain) insertChain(chain types.Blocks, setHead bool) (int, error)
 		if setHead {
 			// First block is pruned, insert as sidechain and reorg only if TD grows enough
 			log.Debug("Pruned ancestor, inserting as sidechain", "number", block.Number(), "hash", block.Hash())
-			return bc.insertSideChain(block, it)
```

Comment:
`insertSideChain` is only called if: (a) parent state is pruned (b) the blocks are inserted via `InsertChain`. 

Given that `InsertChain` is only used in downloader with the semantics of **insert the block as canonical chain**. Therefore, we don't need to bother the witness construction here.

Reply:
I *think* this part was needed so that I could run the state tests for pre-merge networks too? Can't really say for sure, but I kind of remember there was one code path that felt weird to me too, but was needed to allow running the old tests. Maybe not this one?